language: Registry.Register panics on nil receiver

# Summary
`language.Registry.Register` dereferences `r.adapters` without checking whether `r` is nil. That makes the public API inconsistent with `Resolve` and `IDs`, which both guard nil receivers.

# Reproduction steps
1. Run `go test ./internal/language -run TestNilRegistryRegisterPanics -count=1`.
2. Observe the panic from `(*Registry).Register` at `internal/language/registry.go:37`.

# Expected behavior
`Register` should return a normal error for a nil registry receiver, matching the nil-safe behavior already used by `Resolve` and `IDs`.

# Actual behavior
The call panics with `runtime error: invalid memory address or nil pointer dereference`.

# Logs and output
```text
--- FAIL: TestNilRegistryRegisterPanics (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference
github.com/ben-ranford/lopper/internal/language.(*Registry).Register(...)
internal/language/registry.go:37
```

# Version / commit
`bbbb40e5f8b0ff5838379b8e3418ce25a151837f`

# Environment
`go version go1.26.1 darwin/arm64`
`Darwin F4MH3GCYNJ.local 25.3.0 arm64`

--- a/internal/language/registry.go
+++ b/internal/language/registry.go
@@ -24,6 +24,10 @@ func NewRegistry() *Registry {
 }
 
 func (r *Registry) Register(adapter Adapter) error {
+	if r == nil {
+		return errors.New("language registry is nil")
+	}
+
 	if adapter == nil {
 		return errors.New("adapter is nil")
 	}

--- a/internal/language/registry_test.go
+++ b/internal/language/registry_test.go
@@ -126,6 +126,9 @@ func TestResolveAutoTieReturnsError(t *testing.T) {
 
 func TestRegisterValidationAndIDs(t *testing.T) {
 	registry := NewRegistry()
+	if (*Registry)(nil).Register(&testAdapter{id: "nil", detection: Detection{Matched: true}}) == nil {
+		t.Fatalf("expected nil registry register error")
+	}
 	if registry.Register(nil) == nil {
 		t.Fatalf("expected nil adapter error")
 	}

